RecoTracker/TkSeedGenerator/src/SeedingOTEDProducer.cc

```diff
@@ -0,0 +1,321 @@
+#include "RecoTracker/TkSeedGenerator/interface/SeedingOTEDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+#include "RecoTracker/Record/interface/CkfComponentsRecord.h"
+#include "RecoTracker/MeasurementDet/interface/MeasurementTracker.h"
+#include "RecoTracker/MeasurementDet/interface/MeasurementTrackerEvent.h"
+#include "TrackingTools/KalmanUpdators/interface/Chi2MeasurementEstimator.h"
+#include "TrackingTools/PatternTools/interface/TrajectoryStateUpdator.h"
+
+#include "DataFormats/TrajectoryState/interface/LocalTrajectoryParameters.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+#include "TrackingTools/MeasurementDet/interface/TrajectoryMeasurementGroup.h"
+
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+SeedingOTEDProducer::SeedingOTEDProducer(edm::ParameterSet const& conf)
+    : theUpdator(nullptr),
+      tkMeasEventToken(consumes<MeasurementTrackerEvent>(conf.getParameter<edm::InputTag>("trackerEvent"))) {
+  vhProducerToken = consumes<VectorHitCollectionNew>(edm::InputTag(conf.getParameter<edm::InputTag>("src")));
+  beamSpotToken = consumes<reco::BeamSpot>(conf.getParameter<edm::InputTag>("beamSpotLabel"));
+  updatorName = conf.getParameter<std::string>("updator");
+  vhMomHelper = new VectorHitMomentumHelper();
```

Comment:
No need to be a member variable, cheap to construct in the stack of `produce()` (in case the class is kept)

Reply:
Done.